File: app/src/main/java/com/digicoffer/lauditor/Relationships/Adapter/UnshareDocumentAdapter.java
package com.digicoffer.lauditor.Relationships.Adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.fragment.app.FragmentActivity;
import androidx.recyclerview.widget.RecyclerView;

import com.digicoffer.lauditor.Relationships.Model.SharedDocumentsDo;
import com.digicoffer.lauditor.R;
import com.digicoffer.lauditor.CommonFiles.GlobalFiles.DynamicUtils;

import java.util.ArrayList;

public class UnshareDocumentAdapter extends RecyclerView.Adapter<UnshareDocumentAdapter.ViewHolder> {
    ArrayList<SharedDocumentsDo> sharedList = new ArrayList<>();
    ArrayList<SharedDocumentsDo> list_item = new ArrayList<>();
    String Shared_tag = "";
    Context mContext;
    FragmentActivity mActivity;
    RelationshipsAdapter relationshipsAdapter = null;
    String client_id = "";
    String rel_id;

    public UnshareDocumentAdapter(ArrayList<SharedDocumentsDo> sharedList, RelationshipsAdapter relationshipsAdapter) {
        this.relationshipsAdapter = relationshipsAdapter;
        this.sharedList = sharedList;
    }

    public ArrayList<SharedDocumentsDo> getList_item() {
        return sharedList;
    }

    @NonNull
    @Override
    public UnshareDocumentAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.share_document_list, parent, false);
        return new ViewHolder(itemView);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public int getItemViewType(int position) {
        return position;
    }

    @Override
    public void onBindViewHolder(@NonNull UnshareDocumentAdapter.ViewHolder holder, int position) {
        SharedDocumentsDo sharedDocumentsDo = sharedList.get(position);
        holder.tv_share_document.setText(sharedDocumentsDo.getName());
    }


    @Override
    public int getItemCount() {
        return sharedList.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        private TextView tv_share_document;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            {
                tv_share_document = itemView.findViewById(R.id.tv_share_document);
                tv_share_document.setTextSize(DynamicUtils.fifteen);
            }
        }
    }
}

File: app/src/main/java/com/digicoffer/lauditor/Relationships/Adapter/UnshareDocumentAdapter.kt
package com.digicoffer.lauditor.Relationships.Adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.fragment.app.FragmentActivity
import androidx.recyclerview.widget.RecyclerView
import com.digicoffer.lauditor.CommonFiles.GlobalFiles.DynamicUtils
import com.digicoffer.lauditor.R
import com.digicoffer.lauditor.Relationships.Model.SharedDocumentsDo
import java.util.ArrayList

class UnshareDocumentAdapter(
    var sharedList: ArrayList<SharedDocumentsDo>,
    var relationshipsAdapter: RelationshipsAdapter?
) : RecyclerView.Adapter<UnshareDocumentAdapter.ViewHolder>() {

    @get:JvmName("getActualListItem")
    var list_item: ArrayList<SharedDocumentsDo> = ArrayList()
    var Shared_tag: String = ""
    var mContext: Context? = null
    var mActivity: FragmentActivity? = null
    var client_id: String = ""
    var rel_id: String = ""

    fun getList_item(): ArrayList<SharedDocumentsDo> {
        return sharedList
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val itemView = LayoutInflater.from(parent.context).inflate(R.layout.share_document_list, parent, false)
        return ViewHolder(itemView)
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getItemViewType(position: Int): Int {
        return position
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val sharedDocumentsDo = sharedList[position]
        holder.tv_share_document.text = sharedDocumentsDo.name
    }

    override fun getItemCount(): Int {
        return sharedList.size
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val tv_share_document: TextView = itemView.findViewById(R.id.tv_share_document)

        init {
            tv_share_document.textSize = DynamicUtils.fifteen.toFloat()
        }
    }
}
